langchain/utilities/loading.py

```diff
@@ -14,12 +14,50 @@
     "LANGCHAIN_HUB_URL_BASE",
     "https://raw.githubusercontent.com/hwchase17/langchain-hub/{ref}/",
 )
-HUB_PATH_RE = re.compile(r"lc(?P<ref>@[^:]+)?://(?P<path>.*)")
-
+HUB_PATH_RE = re.compile(r"(hf|lc)(?P<ref>@[^:]+)?://(?P<path>.*)")
 
 T = TypeVar("T")
 
 
+def try_load_from_hf_hub(
+    path: Path,
+    loader: Callable[[str], T],
+    **kwargs: Any,
+) -> Optional[T]:
+    """Load configuration from the Hugging Face Hub.
+
+    The Hugging Face Hub automatically has version control, simple sharing mechanism,
+    local caching, and social features (such as likes and Discussions).
+
+    Example:
+        .. code-block:: python
+
+            from langchain.prompts import load_prompt
+            load_prompt("hf://prompts/LLM_Bash/prompt.json")
+    """
+    from huggingface_hub import hf_hub_download
+
+    if len(path.parts) != 3:
+        raise ValueError(
+            "Invalid path. Make sure the path is in the format of "
+            "hf://<suffix>/<repo_id>/<filename>"
+        )
```

Comment:
I guess this is by design but just wanted to mention this means all files have to be stored at the root of the repository (i.e. not possible to do `"hf://prompts/LLM_Bash/stable/prompt.json"`)